operator/pkg/status/kyma_handler.go

```diff
@@ -105,6 +95,20 @@ func (k *Kyma) SyncModuleInfo(kyma *operatorv1alpha1.Kyma, modules parsed.Module
 	}
 }
 
+func (k *Kyma) GetTemplateInfoForModule(
+	kyma *operatorv1alpha1.Kyma,
+	moduleName string,
+) (*operatorv1alpha1.TemplateInfo, error) {
+	for i := range kyma.Status.ModuleInfos {
+		moduleInfo := &kyma.Status.ModuleInfos[i]
+		if moduleInfo.ModuleName == moduleName {
+			return &moduleInfo.TemplateInfo, nil
+		}
+	}
+	// should not happen
```

Comment:
why should this not happen? If there is a mutation done from outside the operator this can happen, right?

Reply:
theoretically it will happen, but nearly impossible, this method happens after `if module.TemplateOutdated` check, and this  TemplateOutdated relays on moduleInfo contains TemplateInfo, if the mutation you mean is manually removed those moduleInfo entry, it will not detect template is outdated.